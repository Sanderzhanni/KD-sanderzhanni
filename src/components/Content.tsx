import React from "react";
import {Link} from "react-router-dom";
import {stories} from "../mocks/stories";

const Content = (): React.ReactElement =>(
  <>
    {stories.map((story, i) =>
      <div key={i} className={"story"}>
        <h3>{story.title}</h3>
        <p>
          {story.text}
        </p>
        <Link to={"TODO"}>Loe lisaks</Link>
      </div>
    )}
  </>
);

export default Content;