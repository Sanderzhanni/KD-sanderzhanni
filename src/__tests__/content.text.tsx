import renderer from "react-test-renderer";
import {BrowserRouter, Link} from "react-router-dom";
import React from "react";
import Content from "../components/Content";
import Enzyme, {shallow, ShallowWrapper} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

const setup = (props={}): ShallowWrapper =>{
  return shallow(<Content {...props}/>);
};

describe("testing header component", () =>{
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Content/>
        </BrowserRouter>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("has a link", ()=>{
    expect(setup().find(Link).length).toBeGreaterThan(0);
  });
});