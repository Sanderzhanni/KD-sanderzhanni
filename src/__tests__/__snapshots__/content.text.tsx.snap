// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing header component renders correctly 1`] = `
Array [
  <div
    className="story"
  >
    <h3>
      LHV loob roheliselt investeeriva hübriidpensionifondi
    </h3>
    <p>
      LHV toob pensioniturule rohelisi ja jätkusuutlikke investeerimispõhimõtteid järgiva pensionifondi. Uus pensionifond kannab nime LHV Pensionifond Roheline ning selle investeerimisportfell hakkab koosnema sotsiaalselt vastutustundlikest, jätkusuutlikest ja keskkonnaalastest investeeringutest. Fondiga võetav aktsiarisk jääb sõltuvalt fondijuhi nägemusest alati 75% ja 100% vahele.
    </p>
    <a
      href="/TODO"
      onClick={[Function]}
    >
      Loe lisaks
    </a>
  </div>,
  <div
    className="story"
  >
    <h3>
      Eriolukorra ajal töötavad kõik LHV pangateenused
    </h3>
    <p>
      Koroonaviiruse leviku tõttu on Eestis kehtestatud eriolukord. Elutähtsa teenuse pakkujana jätkame LHV-s tööd ja anname endast parima, et saaksid pangateenuseid ka praegusel ajal mugavalt kasutada. LHV kliendina saad pea kõiki pangatoiminguid teha kodust lahkumata mobiiliäpi või internetipanga kaudu.
    </p>
    <a
      href="/TODO"
      onClick={[Function]}
    >
      Loe lisaks
    </a>
  </div>,
]
`;
